votepy/ordinal_election.py

```diff
@@ -0,0 +1,57 @@
+from typing import Iterable, Sequence
+
+
+class OrdinalBallot(list):
+    def __init__(self, ordering: Iterable, mapping=None):
+        super().__init__(ordering)
+        
+        self.mapping = mapping
+        
+        for candidate in ordering:
```

Comment:
If you want to allow all types of iterable objects, you need to ensure only one iteration over it. For example, the generator returned by `map` will be depleted after line 6.

Reply:
You should iterate over `self` or shadow variable `ordering`

Reply:
True, fixed